benchmark/distributed/server.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+const yargs = require('yargs');
+const fs = require('fs');
+const path = require('path');
+
+const jstp = require('../..');
+const parseAddress = require('./parse-address');
+
+const args = yargs
+  .option('address', {
+    alias: 'a',
+    describe: 'Address for server to listen on',
+    type: 'string',
+    default: 'tcp://localhost:3030',
+  })
+  .option('master', {
+    alias: 'm',
+    describe: 'Address of master server',
+    type: 'string',
+    default: 'tcp://localhost:3000',
+  })
+  .option('report', {
+    alias: 'r',
+    describe: 'Report interval in ms',
+    type: 'number',
+    default: 10000,
+  })
+  .option('key', {
+    alias: 'k',
+    type: 'string',
+    describe: 'Path to file that contains SSL certificate key',
+    coerce:
+      arg => (arg ? fs.readFileSync(path.resolve(__dirname, arg)) : undefined),
+  })
+  .option('cert', {
+    alias: 'c',
+    type: 'string',
+    describe: 'Path to file that contains SSL certificate',
+    coerce:
+      arg => (arg ? fs.readFileSync(path.resolve(__dirname, arg)) : undefined),
+  })
+  .help()
+  .alias('help', 'h')
+  .argv;
+
+const {
+  address,
+  master,
+  report: reportInterval,
+  key,
+  cert,
+} = args;
+
+const [masterAddressError, parsedMasterAddress] = parseAddress(master);
+if (masterAddressError) {
+  console.error(`Invalid master address: ${masterAddressError.message}`);
+  process.exit(1);
+}
+
+const {
+  transport: masterTransport,
+  connect: masterAddress,
+} = parsedMasterAddress;
+
+const [serverAddressError, parsedServerAddress] = parseAddress(address);
+if (serverAddressError) {
+  console.error(`Invalid server address: ${serverAddressError.message}`);
+  process.exit(1);
+}
+const {
+  transport,
+  listen: listenAddress,
+  connect: connectAddress,
+} = parsedServerAddress;
+
+let timer;
+let startTimeHR = null;
+const application = {
+  server: {
+    method: (connection, argument, callback) => {
+      callback(null, argument);
+    },
+  },
+};
+
+const config = {
+  applications: [new jstp.Application('server', application)],
+  heartbeatInterval: 1000,
+  key,
+  cert,
+};
+
+// Prevent Node.js from throwing DEPTH_ZERO_SELF_SIGNED_CERT.
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+const server = jstp[transport].createServer(config);
+server.maxConnections = 100000;
+
+server.listen(listenAddress, () => {
+  console.log(`Server listening on ${address}`);
+  startTimeHR = process.hrtime();
```

Comment:
Why do you start measuring time here instead of after connection establishment and before `api.master.registerServer`? Or maybe it'll be better to start measuring after `registerServer` but on master-side emit `connect` event after (or setImmediate) calling callback of this method?

Reply:
Because this timer is meant to represent time from passed from master server start.